src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/UnboundGenericType.vb

```diff
@@ -289,6 +289,51 @@ End Class
             Assert.True(DirectCast(c3c6s, INamedTypeSymbol).IsSerializable)
         End Sub
 
+        <Fact>
+        <WorkItem(41779, "https://github.com/dotnet/roslyn/issues/41779")>
+        Public Sub UnboundGenericType_Bug41779()
+
+            Dim compilation = CompilationUtils.CreateCompilation(
+<compilation name="C">
+    <file name="a.vb"><![CDATA[
+Imports System.Runtime.CompilerServices
+
+Interface I
+    Function GetService() As Object
+End Interface
+
+Module Program
+    <Extension()>
+    Private Function GetService(Of T)(ByVal obj As I) As T
+        Return "default"
+    End Function
+
+    Private Sub M(ByVal provider As I)
+        provider.GetService(Of)()
+        provider.GetService(Of)().ToString()
+        provider.GetService(Of)()
+    End Sub
+End Module
+    ]]></file>
+</compilation>)
+
+            compilation.AssertTheseDiagnostics(<expected>
+BC30311: Value of type 'String' cannot be converted to 'T'.
+        Return "default"
+               ~~~~~~~~~
+BC30182: Type expected.
+        provider.GetService(Of)()
+                              ~
+BC30182: Type expected.
+        provider.GetService(Of)().ToString()
+                              ~
+BC30182: Type expected.
+        provider.GetService(Of)()
+                              ~
+                </expected>)
+
+        End Sub
```

Comment:
>End Sub [](start = 8, length = 7)

I recommend using the same test matrix as we end up with for C#.

Reply:
As in using all the same tests (translated to VB), or using the same testing methodology?

Reply:
> As in using all the same tests (translated to VB), or using the same testing methodology?

I would port all the same tests. But let's wait until we are good on C# side.

---
In reply to: [442359826](https://github.com/dotnet/roslyn/pull/45033#discussion_r442359826) [](ancestors = 442359826)